src/dataExplorer/components/ResultsPane.tsx

```diff
@@ -119,71 +119,61 @@ const ResultsPane: FC = () => {
   }
 
   const downloadByServiceWorker = () => {
-    if (language === LanguageType.INFLUXQL) {
-      try {
-        event('runQuery.downloadCSV.influxql', {context: 'query experience'})
+    event(`runQuery.downloadCSV`, {context: `query experience: ${language}`})
 
-        const params = new URLSearchParams({
-          db: selection.dbrp?.database,
-          q: text,
-        })
-        if (selection.dbrp?.retention_policy) {
-          params.set('rp', selection.dbrp?.retention_policy)
-        }
-        const url = `${API_BASE_PATH}query?${params}`
-        const hiddenForm = document.createElement('form')
-        hiddenForm.setAttribute('id', 'downloadDiv')
-        hiddenForm.setAttribute('style', 'display: none;')
-        hiddenForm.setAttribute('method', 'post')
-        hiddenForm.setAttribute('action', url)
+    try {
+      let url: string = ''
+      let inputValue: string = ''
 
-        const input = document.createElement('input')
-        input.setAttribute('name', 'data')
-        input.setAttribute('value', '')
-        hiddenForm.appendChild(input)
-        document.body.appendChild(hiddenForm)
-        hiddenForm.submit()
-      } catch (error) {
-        dispatch(notify(csvDownloadFailure()))
+      switch (language) {
+        case LanguageType.FLUX:
+        case LanguageType.SQL:
+          url = `${API_BASE_PATH}api/v2/query?${new URLSearchParams({orgID})}`
+          const query =
+            language == LanguageType.SQL
+              ? sqlAsFlux(text, selection.bucket)
```

Comment:
what is the purpose of this ternary where its not lang type sql? We'd ever hit this anymore given the switch statement determines that block language type SQL

Reply:
Yes, this ternary will be hit because SQL and Flux are sharing the same case statement here. Basically, SQL query is wrapped in a Flux query, see [sqlAsFlux](https://github.com/influxdata/ui/blob/0266f8212eddf0cfbb65551af67b74539d081bce/src/shared/contexts/query/preprocessing.ts#L420-L431) function. This will change after the backend work for [removing Flux from non-Flux query](https://github.com/influxdata/idpe/issues/17711) is done.

Reply:
LanguageType.SQL would fall into this code block so then LanguageType.SQL ? will always evaluate to sqlAsFlux in which case what is the point of the else text i guess is my real question?

Reply:
oh i see it now sounds good